integration-tests/ci-visibility/jest-custom-test-sequencer.js

```diff
@@ -0,0 +1,21 @@
+const Sequencer = require('@jest/test-sequencer').default
+
+// From example in https://jestjs.io/docs/configuration#testsequencer-string
+class CustomSequencer extends Sequencer {
+  shard (tests, { shardIndex, shardCount }) {
+    // Log used to show that the custom sequencer is being used
+    // eslint-disable-next-line
+    console.log('Running shard with a custom sequencer', shardIndex)
+    const shardSize = Math.ceil(tests.length / shardCount)
+    const shardStart = shardSize * (shardIndex - 1)
+    const shardEnd = shardSize * shardIndex
+
+    return [...tests].sort((a, b) => (a.path > b.path ? 1 : -1)).slice(shardStart, shardEnd)
```

Comment:
Looks like for the last shard the calculated `shardEnd` might lie outside the boundaries of the `tests` archive. Can the `slice` method handle this?

Reply:
good catch, it can indeed.